clients/unshaded/src/main/java/tachyon/client/file/FileOutStream.java

```diff
@@ -97,8 +100,8 @@ public FileOutStream(long fileId, OutStreamOptions options) throws IOException {
     }
     mClosed = false;
     mCanceled = false;
-    mHostname = options.getHostname();
     mShouldCacheCurrentBlock = mTachyonStorageType.isStore();
+    mLocationPolicy = FileWriteLocationPolicy.Factory.createPolicy(new TachyonConf(), options);
```

Comment:
Instead of passing the entire `OutputStreamOptions` into the `FileWriteLocationPolicy`, could we have a `FileWriteLocationPolicyOptions` field in `OutputStreamOptions` which is used to initialize the `FileWriteLocationPolicy`? Then e.g. `DefaultLocationPolicy` could be configured by a `DefaultLocationPolicyOptions` which implements `FileWriteLocationPolicyOptions`